collect_app/src/test/java/org/odk/collect/android/widgets/range/RangePickerDecimalWidgetTest.kt

```diff
@@ -0,0 +1,160 @@
+package org.odk.collect.android.widgets.range
+
+import android.view.View.OnLongClickListener
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.MatcherAssert
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers
+import org.hamcrest.Matchers.equalTo
+import org.hamcrest.Matchers.nullValue
+import org.javarosa.core.model.RangeQuestion
+import org.javarosa.core.model.data.StringData
+import org.javarosa.form.api.FormEntryPrompt
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.odk.collect.android.R
+import org.odk.collect.android.formentry.questions.QuestionDetails
+import org.odk.collect.android.widgets.support.QuestionWidgetHelpers
+import java.math.BigDecimal
+
+@RunWith(AndroidJUnit4::class)
+class RangePickerDecimalWidgetTest {
+    private val rangeQuestion: RangeQuestion = RangeQuestion().also {
+        it.rangeStart = BigDecimal(1.5)
+        it.rangeEnd = BigDecimal(5.5)
+        it.rangeStep = BigDecimal(0.5)
+    }
+
+    @Test
+    fun `list of numbers should contain only rangeStart when range has equal start and end`() {
+        rangeQuestion.rangeStart = BigDecimal(5.0)
+        rangeQuestion.rangeEnd = BigDecimal(5.0)
+        rangeQuestion.rangeStep = BigDecimal(1.0)
+
+        val widget = createWidget(QuestionWidgetHelpers.promptWithQuestionDefAndAnswer(rangeQuestion, null))
+
+        assertThat(widget.displayedValuesForNumberPicker, equalTo(arrayOf("5.0")))
```

Comment:
Instead of needing to expose a private field from `RangePickerXWidget`, could we just have these tests targeting the `NumberPickerDialog` and pass it the start, end and step? Alternatively, I'd also be happy with just testing `RangePickerWidgetUtils.getNumbersFromRangeAsc` directly.

Reply:
Ok, I implemented it in that second way.